src/backend/joanie/tests/core/test_api_contract.py

```diff
@@ -621,3 +629,424 @@ def test_api_contract_download_authenticated_should_fail_if_contract_is_not_sign
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+        self.assertEqual(response.json(), ["No zip to generate"])
+
+    # pylint: disable=too-many-locals
+    def test_api_contract_generate_zip_archive_authenticated_post_method_allowed(self):
+        """
+        Authenticated user should be able to use POST method on the viewset to generate ZIP
+        archive when parsing an existing Organization UUID where the user has the rights to access,
+        and it will generate a ZIP archive with the signed contracts.
+        """
+        storage = storages["contracts"]
+        requesting_user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(
+            organization=organization, user=requesting_user
+        )
+        relation = factories.CourseProductRelationFactory(organizations=[organization])
+        learners = factories.UserFactory.create_batch(3)
+        signature_reference_choices = [
+            "wfl_fake_dummy_1",
+        ]
+        for index, reference in enumerate(signature_reference_choices):
+            user = learners[index]
+            order = factories.OrderFactory(
+                owner=user,
+                product=relation.product,
+                course=relation.course,
+                state=enums.ORDER_STATE_VALIDATED,
+            )
+            context = contract_definition.generate_document_context(
+                order.product.contract_definition, user, order
+            )
+            factories.ContractFactory(
+                order=order,
+                signature_backend_reference=reference,
+                definition_checksum="1234",
+                context=context,
+                signed_on=timezone.now(),
+            )
+        expected_endpoint_polling = "/api/v1.0/contracts/zip_archive/"
+        token = self.get_user_token(requesting_user.username)
+
+        response = self.client.post(
+            "/api/v1.0/contracts/zip_archive/",
+            data={"organization_id": organization.id},
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 202)
+
+        content = response.content.decode("utf-8")
+        content_json = json.loads(content)
+        polling_url = content_json["url"]
+        generated_zip_uuid = polling_url[-37:-1]
+
+        self.assertEqual(
+            content_json["url"], f"{expected_endpoint_polling}{generated_zip_uuid}/"
+        )
+        self.assertEqual(len(generated_zip_uuid), 36)
+        self.assertTrue(
+            storage.exists(f"{requesting_user.id}_{generated_zip_uuid}.zip")
+        )
+
+    def test_api_contract_get_zip_archive_anonymous(self):
+        """
+        Anonymous user should not be able to get ZIP archive.
+        """
+        response = self.client.get(
+            reverse("contracts-zip_archive", kwargs={"zip_id": uuid4()})
```

Comment:
I prefer using a real endpoint path here.

```suggestion
            f"/api/v1.0/contracts/zip_archive/{uuid4()}"
```

Reply:
I agree but this is not how it is made in joanie tests. I did it for a seek of consistency